src/interfaces/event-interfaces.ts

```diff
@@ -0,0 +1,46 @@
+import { IntMenuProvider, IntMenuShortcut } from './menu-interfaces';
+
+/** Toggles Menu && Sets open panel */
+export interface IntManageSideMenuEvent extends CustomEvent {
+  type: 'updateSideMenu';
+  detail: {
+    menuId: string | undefined | '';
+    action: 'open' | 'toggle' | '';
+  };
+}
+
+/** Sets open panel */
+export interface IntSetOpenMenuEvent extends CustomEvent {
+  type: 'menuTypeSelected';
+  detail: {
+    id: string /** name of menu to open */;
```

Comment:
Not sure if you have an enum of menus to open, but if possible, I would make this limited to a list of potential options instead of a `string`.